core/src/main/java/kafka/server/handlers/DescribeTopicPartitionsRequestHandler.java

```diff
@@ -103,14 +102,20 @@ public DescribeTopicPartitionsResponseData handleDescribeTopicPartitionsRequest(
                 );
             }
             return isAuthorized;
-        }).sorted().iterator();
+        }).sorted();
+
+        // Reset the first partition index if the cursor topic is missing from the authorized topic list.
+        int firstPartitionId = !cursorTopicName.isEmpty() && authHelper.authorize(
+            abstractRequest.context(), DESCRIBE, TOPIC, cursorTopicName, true, true, 1)
+            ? cursor.partitionIndex() : 0;
```

Comment:
We'd need to handle both non-authorized and deleted cursor topic.  If we could peek if the first item in the authorized stream is equal to cursor topic, we can handle both.

If we follow @mumrah's suggestion, we wouldn't have the sorted stream anymore, in that case, we should check if partitionIndex >0 && !currentTopic.equals(cursorTopic) then use partitionIndex 0 (in other words, non-0 partition index can only be applied if we are processing cursor topic (which would be the first topic, unless it gets deleted or unauthorized).

Reply:
Good point. Previously I did not find an easy way to get the first element of a stream/iterator, so I verify the cursor topic again. Now I passed a functor to return the start index for each topic which only return non zero value for cursor topic.